tests/lean/replaceLocalDeclInstantiateMVars.lean

```diff
@@ -0,0 +1,53 @@
+import Lean
+
+/-!
+# Ensure `replaceLocalDecl` instantiates metavariables
+
+These tests ensure that `replaceLocalDecl` does not introduce unknown `FVarId`s by inserting a new
+local declaration too early.
+
+## Context
+
+`replaceLocalDecl mvarId fvarId typeNew eqProof` replaces `fvarId` with a new FVarId that has the
+same name but type `typeNew`. It does this by inserting a new local declaration with type `typeNew`
+and clearing the old one if possible.
+
+It makes sure that the new local declaration is inserted at the soonest point after `fvarId` at
+which `typeNew` is well-formed. It does this by traversing `typeNew` and finding the `FVarId` with
+the maximum index among all `FVarId`s occurring in `typeNew`.
+
+However, if it encounters a metavariable during this traversal, it simply continues traversing,
+even if this metavariable is assigned to an expression which contains an `FVarId` occurring after
+`fvarId`. This can lead to `replaceLocalDecl` inserting a local declaration with a type which
+depends on `FVarId`s which come after it in the local context, thus creating unknown `FVarId`s (the
+`FVarId`s of the local declarations after the insertion are modified).
```

Comment:
Can you clarify that this is describing the previous behaviour (which would have caused this test to fail)?

Reply:
Okay, done in fbc8de6.